redbeams-api/src/main/java/com/sequenceiq/redbeams/validation/DatabaseVendorAndServiceValidator.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.redbeams.validation;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.validation.ValidatorUtil;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedDatabaseEntry;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedDatabaseProvider;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedExternalDatabaseServiceEntry;
+import com.sequenceiq.redbeams.api.endpoint.v4.database.request.DatabaseV4Request;
+import com.sequenceiq.redbeams.api.util.DatabaseVendorUtil;
+
+public class DatabaseVendorAndServiceValidator implements ConstraintValidator<ValidDatabaseVendorAndService, DatabaseV4Request> {
+
+    private DatabaseVendorUtil databaseVendorUtil = new DatabaseVendorUtil();
+
+    @Override
+    public boolean isValid(DatabaseV4Request request, ConstraintValidatorContext context) {
+        Optional<DatabaseVendor> vendorByJdbcUrl = databaseVendorUtil.getVendorByJdbcUrl(request);
+        if (!vendorByJdbcUrl.isPresent()) {
+            ValidatorUtil.addConstraintViolation(context, "Could not determine database vendor from JDBC URL "
+                + request.getConnectionURL(), "connectionURL");
+            return false;
+        }
+
+        Optional<SupportedExternalDatabaseServiceEntry> serviceEntry = SupportedDatabaseProvider.supportedExternalDatabases()
+                .stream()
+                .filter(item -> item.getName().equalsIgnoreCase(request.getType())
+                        || item.getDisplayName().equalsIgnoreCase(request.getType()))
+                .findFirst();
+        if (!serviceEntry.isPresent()) {
+            serviceEntry = SupportedDatabaseProvider.getOthers();
```

Comment:
Another idea for a small refactoring (not so important): getOthers() method, instead of a search, could return a static stored default element called other. Also, we could assume that getOther() always returns the default entry (so it is not an optional one), and then two lines below we do not have to check again if it is present. But really, this is just an idea, maybe not worth the effort.